python/semantic_kernel/processes/kernel_process/kernel_process_step_state_metadata.py

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from typing import Any, Literal
+
+from pydantic import Field
+
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+
+
+class KernelProcessStepStateMetadata(KernelBaseModel):
+    """Process state used for State Persistence serialization."""
+
+    type_: Literal["Step", "Process"] = Field("Step", alias="$type")
+    id: str | None = Field(None, alias="id")
+    name: str | None = Field(None, alias="name")
+    version_info: str | None = Field(None, alias="versionInfo")
+    state: Any | None = Field(None, alias="state")
```

Comment:
How do we ensure this is json serializable?

Reply:
Updated to `state: TState` which is bound as a base model.